frontend/src/pages/frontPage/FrontPage.jsx

```diff
@@ -1,16 +1,84 @@
-import styling from "./panel.module.css";
+import styling from "./frontpage.module.css";
 import panelImg from "../../assets/images/panelImage.jpg";
+import pencil from "../../assets/images/brokenPencil.png";
+import { useEffect } from "react";
+import { getCourses } from "../../services.js/api";
+import { useState } from "react";
+
 
 const FrontPage = () => {
-    return(
-        <>
-            <div className={styling.container}>
-                <img id={styling.panel} src={panelImg}/>
-                <div className={styling.panelText}>
-                    <h3>Join LearningHub to manage or inspire your students effortlessly or sign up and start learning the fundamentals of coding </h3>
+    const [courseDescriptions, setCoursesDescriptions] = useState([]);
+    const [state, setState] = useState({
+        loading: true,
+        error: false
+    });
+
+    useEffect(() => {
+        getCourses()
+        .then(data => {
```

Comment:
nit: generally prefer `async/await` syntax to promises.

Reply:
will switch as it is better practice I've noticed. Thank you.